site/docs/pages/swap/swap.mdx

```diff
@@ -90,7 +93,7 @@ export default function SwapComponents() {
         token={USDCToken} // [!code focus]
         type="to" // [!code focus]
       /> // [!code focus]
-      <SwapButton onError={onError} onSubmit={onSubmit} /> // [!code focus]
+      <SwapButton onError={onError} onSubmit={onSubmit} disabled={isDisabled} /> // [!code focus]
```

Comment:
We might want to handle the disable state internally and not expose it externally.

The `SwapButton` should only be enabled if 
- `fromAmount`, `fromToken`, `toToken` is set
- AND user has the available balance on their wallet

Reply:
After discussion, we need to expose it externally to explicitly disable the button

Reply:
shouldn't the developer have maximum control of the component?

e.g. `shadcn` is a library we want to base `OnchainKit` off, and they pass in `disabled` as a prop to their `Button` component ([example](https://ui.shadcn.com/docs/components/button#loading))

Reply:
I +1 exposing control to disable how it is currently written! We can apply additional automatic disabled states for things like missing inputs, but enabling apps to have more control is our strategy seemingly!

Reply:
All those points are valid point! We are carefully considering what should be exposed and what shouldn't. `disable` prop on the `SwapButton` is very reasonable and we should do it. :)